src/repos/ClubRepo.ts

```diff
@@ -0,0 +1,37 @@
+import { getConnectionManager, Repository } from 'typeorm';
+import Club from '../entities/Club';
+
+const db = (): Repository<Club> =>
+  getConnectionManager()
+    .get()
+    .getRepository(Club);
+
+const createClub = async (name: string): Promise<void> => {
+  const possibleClub = await db().findOne({ name });
+  if (!possibleClub) {
+    const club = db().create({
+      name,
+      users: [],
+    });
+    await db().save(club);
+  }
+  return;
+};
+
+const getClubByName = async (name: string): Promise<Club> => {
+  const club = await db().findOne({ where: { name } });
+  if (!club) {
+    throw Error('Club with that name not found');
```

Comment:
are you sure you want to throw an Error? Or do you just want to return nothing? will the client in the end just get an empty array or do you just send them a 404

Reply:
Thanks for pointing that out but I think throwing an error works best here.